### PR TITLE
Fix build with ip-1.5 and later

### PR DESCRIPTION
**Pull request by hasufell**
This fixes some pesky issues with cabal in other cardano repos.

### CODE REVIEW

**Comment by hasufell** [comment]
@KtorZ

File: cardano-sl-x509.cabal
Comment (hasufell) [inline cardano-sl-x509.cabal:42]: would be nice to be conditional, but Cabal file format isn't expressive enough. Isn't a problem since it's in stackage anyway.

File: src/Data/X509/Extra.hs
Comment (hasufell) [inline src/Data/X509/Extra.hs:230]: commit is here https://github.com/andrewthad/haskell-ip/commit/0742cd48f5a49a0e8718f6eb559c70d5192ac799
changelog here https://github.com/andrewthad/haskell-ip/blob/master/changelog.md#150---2019-03-23

**Review by KtorZ — APPROVED** [review]
LGTM although there's no automated CI on this repository.

@hasufell , can you confirm that the tests are passing on your local branch before merging this?

**Comment by hasufell** [comment]
Tested with cabal (`ip-1.7.2`) and stack (`ip-1.4.2.1`)

**Comment by hasufell** [comment]
I don't think I have write access to this repo.

**Comment by KtorZ** [comment]
you should now. :+1:

**Pull request merged**
